cdap-tms-spanner/src/main/java/io.cdap.cdap.messaging/SpannerMessagingService.java

```diff
@@ -0,0 +1,205 @@
+import io.cdap.cdap.messaging.spi.RawMessage;
+import io.cdap.cdap.messaging.spi.RollbackDetail;
+import io.cdap.cdap.messaging.spi.StoreRequest;
+import io.cdap.cdap.messaging.spi.TopicMetadata;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.proto.id.TopicId;
+import io.cdap.cdap.security.spi.authorization.UnauthorizedException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutionException;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The Spanner based implementation of {@link MessagingService}.
+ */
+public class SpannerMessagingService implements MessagingService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SpannerMessagingService.class);
+
+  private Map<String, String> cConf;
+
+  private Spanner spanner;
+
+  private DatabaseClient client;
+
+  private DatabaseAdminClient adminClient;
+
+  private final ConcurrentLinkedQueue<StoreRequest> batch = new ConcurrentLinkedQueue<>();
+
+  public static final String PAYLOAD_FIELD = "payload";
+  public static final String PUBLISH_TS_FIELD = "publish_ts";
+  public static final String PAYLOAD_SEQUENCE_ID = "payload_sequence_id";
+  public static final String SEQUENCE_ID_FIELD = "sequence_id";
+
+  @Override
+  public void initialize(MessagingContext context) throws IOException {
+    this.cConf = context.getProperties();
+    this.spanner = SpannerUtil.getSpannerService(cConf);
+    this.client = SpannerUtil.getSpannerDbClient(cConf, spanner);
+    this.adminClient = SpannerUtil.getSpannerDbAdminClient(spanner);
+    LOG.info("Spanner messaging service started.");
+  }
+
+  @Override
+  public String getName() {
+    return this.getClass().getSimpleName();
+  }
+
+  @Override
+  public void createTopic(TopicMetadata topicMetadata)
+      throws TopicAlreadyExistsException, IOException, UnauthorizedException {
+    createTopic(topicMetadata.getTopicId());
+  }
+
+  private void createTopic(TopicId topicId) {
+    String topicSQL = String.format("CREATE TABLE IF NOT EXISTS %s ( %s INT64, %s INT64, %s"
+            + " TIMESTAMP NOT NULL OPTIONS (allow_commit_timestamp=true), %s BYTES(MAX) )"
+            + " PRIMARY KEY (sequence_id, payload_sequence_id, publish_ts), ROW DELETION POLICY"
+            + " (OLDER_THAN(publish_ts, INTERVAL 7 DAY))", getTableName(topicId), SEQUENCE_ID_FIELD,
+        PAYLOAD_SEQUENCE_ID, PUBLISH_TS_FIELD, PAYLOAD_FIELD);
+    OperationFuture<Void, UpdateDatabaseDdlMetadata> future = adminClient.updateDatabaseDdl(
+        SpannerUtil.getInstanceID(cConf), SpannerUtil.getDatabaseID(cConf),
+        Collections.singletonList(topicSQL), null);
+    try {
+      future.get();
+    } catch (InterruptedException | ExecutionException e) {
+      LOG.error("Error when executing {}", topicSQL, e);
```

Comment:
We should not log and forget the exception. Check the handleError in ClientMessagingService